os/arch/arm/src/armv7-a/arm_gicv2.c

```diff
@@ -508,6 +508,43 @@ int up_prioritize_irq(int irq, int priority)
 	return -EINVAL;
 }
 
+/****************************************************************************
+ * Name: up_affinity_irq
+ *
+ * Description:
+ *   set the affinity of a given irq (SPI only) to given cpu.
+ *   Currenlty, it doesnot support assigning multiple cpus as affinity.
+ *   It can assign only one cpu per SPI.
+ *
+ * Input Parameters:
+ *   irq: irq id
+ *   cpu: cpu id
+ *
+ * Returned Value:
+ *   None
+ *
+ ****************************************************************************/
+
+void up_affinity_irq(int irq, int cpu)
```

Comment:
This api can be easily changed to handle multiple cpu affinity. Instead of using `int cpu` let us use `cpu_set_t`. The same is used for setting affinity of tasks. The cpu set data type will have one bit for each cpu. If the bit is set, then the irq can be taken to that particular cpu.